test/include/graph_test/graph_test.h

```diff
@@ -32,6 +37,35 @@ class EmptyDBTest : public PrivateGraphTest {
     std::string getInputDir() override { KU_UNREACHABLE; }
 };
 
+class ConcurrentTest {
+public:
+    ConcurrentTest(bool& connectionsPaused, main::Connection& connection, std::string databasePath):
+      connectionPaused{connectionsPaused}, connection{connection}, databasePath{databasePath} {}
+    void execute() {
+        connThread = std::thread(&ConcurrentTest::runStatements, this);
+    }
+    void join() {
+        if (connThread.joinable()) {
+            connThread.join();
+        }
+    }
+    void reset() {
+        statementsQueue.clear();
+    }
+    void addStatement(TestStatement* statement) {
+        statementsQueue.emplace_back(statement);
+    }
+
+private:
+    void runStatements();
+
+    bool& connectionPaused;
```

Comment:
`connectionPaused` is also not needed if you don't busy wait inside `runStatements`.